lib/plugins/inspec-habitat/lib/inspec-habitat/profile.rb

```diff
@@ -1,79 +1,96 @@
-        build_hart
+        @log.info("Creating a Habitat artifact for '#{@path}'")
+        working_dir = create_working_dir
+        habitat_config = read_habitat_config
+
+        output_dir = @options[:output_dir] || Dir.pwd
+        unless File.directory?(output_dir)
+          exit_with_error("Output directory #{output_dir} is not a directory " \
+                          'or does not exist.')
+        end
+
+        verify_habitat_setup(habitat_config)
+
+        duplicated_profile = duplicate_profile(@path, working_dir)
+        prepare_profile!(duplicated_profile)
+
+        hart_file = build_hart(working_dir, habitat_config)
+
+        @log.info("Copying artifact to #{output_dir}...")
+        destination = File.join(output_dir, File.basename(hart_file))
+        FileUtils.cp(hart_file, destination)
+
+        destination
       rescue => e
         @log.debug(e.backtrace.join("\n"))
         exit_with_error(
-          'Unable to generate Habitat artifact.',
+          'Unable to create Habitat artifact.',
           "#{e.class} -- #{e.message}",
         )
+      ensure
+        @log.debug("Deleting working directory #{working_dir}")
+        FileUtils.rm_rf(working_dir)
       end
 
-      def copy(hart_file)
-        validate_output_dir
+      def setup(profile = profile_from_path(@path))
+        path = profile.root_path
+        @log.info("Setting up #{path} for Habitat...")
 
-        @log.info("Copying artifact to #{output_dir}...")
-        copy_hart(hart_file)
+        plan_file = File.join(path, 'habitat', 'plan.sh')
+        @log.info("Generating Habitat plan at #{plan_file}...")
+        vars = {
+          profile: profile,
+          habitat_origin: read_habitat_config['origin'],
+        }
+        create_file_from_template(plan_file, 'plan.sh.erb', vars)
+
+        run_hook_file = File.join(path, 'habitat', 'hooks', 'run')
+        @log.info("Generating a Habitat run hook at #{run_hook_file}...")
+        create_file_from_template(run_hook_file, 'hooks/run.erb')
+
+        default_toml = File.join(path, 'habitat', 'default.toml')
+        @log.info("Generating a Habitat default.toml at #{default_toml}...")
+        create_file_from_template(default_toml, 'default.toml.erb')
+
+        config = File.join(path, 'habitat', 'config', 'inspec_exec_config.json')
+        @log.info("Generating #{config} for `inspec exec`...")
+        create_file_from_template(config, 'config/inspec_exec_config.json.erb')
       end
 
       def upload
-        validate_habitat_auth_token
-        hart_file = create
-        upload_hart(hart_file)
+        habitat_config = read_habitat_config
+
+        if habitat_config['auth_token'].nil?
+          exit_with_error(
+            'Unable to determine Habitat auth token for uploading.',
+            'Run `hab setup` or set the HAB_AUTH_TOKEN environment variable.',
+          )
+        end
+
+        # Run create command to create habitat artifact
+        hart = create
```

Comment:
If the `create` function is only every called from within this class, could we rename it to `create_hart`?

For public methods, something like `create` can make sense because when called on a class it shows purpose (`habitat_profile.create`). But for private or internally used methods I like more verbosity to easily understand purpose.

Style comment though - feel free to disagree

Reply:
It's called here https://github.com/inspec/inspec/blob/master/lib/plugins/inspec-habitat/lib/inspec-habitat/cli.rb#L17